internal/parse/parser_test.go

```diff
@@ -12,3 +12,73 @@ func TestInit(t *testing.T) {
 	assert.Equal(t, nil, err)
 	assert.Equal(t, (*ParsedExpr)(nil), expr)
 }
+
+func TestPeekByte(t *testing.T) {
+	p := NewParser()
+	p.Parse("")
+	assert.Equal(t, false, p.peekByte('a'))
```

Comment:
Might be worth using go-check, which allows us to have suites and is what we use the most at Canonical. I can provide some simple samples if needed.  Okay to wait a bit until it's more obvious, if you feel like it.